app/assets/javascripts/modal.js

```diff
@@ -0,0 +1,21 @@
+var modal = document.getElementById('saveModal')
```

Comment:
minor feedback, but I would consider wrapping all the code in this file with an IIFE (https://developer.mozilla.org/en-US/docs/Glossary/IIFE) so the variables don't become globally accessible. one easy way to prevent future name collisions.